### PR TITLE
Merge qdq and quant-to-int passes

### PR DESCRIPTION
**Pull request by sdasgup3**
This PR does the followings:
1. Merges `stablehlo-legalize-quantized-op-to-qdq` into  `stablehlo-legalize-quant-to-int`. 
1. Rename `stablehlo-legalize-quant-to-int` to `stablehlo-legalize-quant-to-math`. This is to clarify for scenario when the fallback `qdq` is used and sull integer quantized program cannot be generated. 
1. Removes  `stablehlo-legalize-quantized-op-to-qdq` pass and replace its uses with `stablehlo-legalize-quant-to-math`. 
1. Remove QDQ lit checks from `stablehlo/tests/ops_stablehlo_quantized.mlir` and merges the tests added for qdq pass in  `stablehlo/tests/stablehlo_legalize_quant_to_int.mlir`
1. Updates the tests in `stablehlo/tests/stablehlo_legalize_quant_to_int.mlir` updating  __only__ negatives tests, which are previously unhanded by `stablehlo-legalize-quant-to-int`. The current `stablehlo-legalize-quant-to-math` uses the fallback to handle these cases. 
1. About the pass    `stablehlo-legalize-quant-to-math`
  - It uses `Patternbenefit` to assign highest priority (`benefit=10`) to pattern which has specialized handling in `stablehlo-legalize-quant-to-int`. Next in priority (`benefit=0`) are the QDQ patterns.  
 
With that the following program, which `stablehlo-legalize-quant-to-int` has specialized handling, will avoid the fallback path. 
```
func.func @max_per_tensor_same_quant_parameters(
    %arg0: tensor<1x2x!quant.uniform<i8:f32, 2.000000e+00:3>>
  ) -> tensor<1x2x!quant.uniform<i8:f32, 2.000000e+00:3>> {
  %0 = "stablehlo.maximum"(%arg0, %arg0) : (
    tensor<1x2x!quant.uniform<i8:f32, 2.000000e+00:3>>,
    tensor<1x2x!quant.uniform<i8:f32, 2.000000e+00:3>>
  ) -> tensor<1x2x!quant.uniform<i8:f32, 2.000000e+00:3>>
  return %0 : tensor<1x2x!quant.uniform<i8:f32, 2.000000e+00:3>>
}
```

whereas the following, which is not supported in `stablehlo-legalize-quant-to-int` will choose the fallback path.
```
func.func @max_per_tensor_diff_quant_parameters(%arg0: tensor<!quant.uniform<i8:f32,1.0:0>>, %arg1: tensor<!quant.uniform<i8:f32,2.0:1>>) ->  tensor<!quant.uniform<i8:f32,3.0:2>> {
  %0 = "stablehlo.maximum"(%arg0, %arg1) : (tensor<!quant.uniform<i8:f32,1.0:0>>, tensor<!quant.uniform<i8:f32,2.0:1>>) -> tensor<!quant.uniform<i8:f32,3.0:2>>
  func.return %0 : tensor<!quant.uniform<i8:f32,3.0:2>>
}
```

  - Currently handles qdq fallback for AddOp and a bunch of `GenericOps` op [cs](https://github.com/openxla/stablehlo/blob/eba821aa1c54a21d70331d7926dfc8b929f988f3/stablehlo/transforms/StablehloLegalizeQuantToInt.cpp#L1239). qdq fallback for `dot_general` and `convolution` will be handled in a follow up PR. What this means is we will still see  quantized dot_gneral/convolution program, which are currently unsupported by  `stablehlo-legalize-quant-to-int`, error out. 
 

[childPR](https://github.com/openxla/stablehlo/pull/2459)

### CODE REVIEW

**Review by GleasonK — CHANGES_REQUESTED** [review]
One comment, otherwise LGTM!

File: stablehlo/transforms/StablehloLegalizeQuantizedOpToQDQ.cpp [hunk 1]
```diff
@@ -1,136 +0,0 @@
...
-==============================================================================*/
-
-#include "mlir/Dialect/Func/IR/FuncOps.h"
-#include "mlir/Dialect/Quant/QuantTypes.h"
-#include "mlir/IR/BuiltinOps.h"
-#include "mlir/IR/PatternMatch.h"
-#include "mlir/Pass/Pass.h"
-#include "mlir/Transforms/DialectConversion.h"  // Include for TypeConverter
-#include "mlir/Transforms/GreedyPatternRewriteDriver.h"
-#include "stablehlo/dialect/StablehloOps.h"
```
[the remark this answers is not in the document]
Comment (sdasgup3) [reply stablehlo/transforms/StablehloLegalizeQuantizedOpToQDQ.cpp]: make sense. Done!
Also, the lit tests checking the qdq pass are moved from `ops_stablehlo_quantized.mlir`  to a dedicated file `stablehlo_legalize_quantized_op_to_qdq.mlir`. The checks in  `ops_stablehlo_quantized.mlir` are only for quantized IR IR verification.

File: stablehlo/transforms/StablehloLegalizeQuantizedOpToQDQ.cpp [hunk 2]
```diff
@@ -109,27 +109,28 @@ class StablehloLegalizeQuantizedOpToQDQPass
...
 }  // namespace
 
 void populateStablehloLegalizeQuantizedOpToQDQPatterns(
-    RewritePatternSet* patterns, MLIRContext* context) {
+    RewritePatternSet* patterns, MLIRContext* context, PatternBenefit benefit) {
   // The following list covers most of the operations which, according to the
   // stablehlo spoecification document, interprets the quantized
   // operation using dequant-op-quant strategy. The ones excluded are
-  // AddOP, ConvolutionOp, DotGeneralOp, and DynamicConvOp, which are current
+  // ConvolutionOp, DotGeneralOp, and DynamicConvOp, which are current
```
Comment (GleasonK) [inline stablehlo/transforms/StablehloLegalizeQuantizedOpToQDQ.cpp:124]: We're using benefit to manage this now, can we delete this comment and add these ops into the pass + tests? (Can be done in a followup)

**Review by GleasonK — APPROVED** [review]

**Review by abhigunj — APPROVED** [review]

File: stablehlo/transforms/StablehloLegalizeQuantizedOpToQDQ.cpp (hunk 2 above)
Comment (sdasgup3) [reply stablehlo/transforms/StablehloLegalizeQuantizedOpToQDQ.cpp]: make sense. Remove the comment. Will make sure any ops added to the list has the requisite test-cases.

**Pull request merged**